components/script/dom/htmlanchorelement.rs

```diff
@@ -566,6 +568,64 @@ impl Activatable for HTMLAnchorElement {
     }
 }
 
+/// <https://html.spec.whatwg.org/multipage/semantics.html#get-an-element's-target>
+pub fn get_element_target(subject: &Element) -> Option<DOMString> {
+    if !(subject.is::<HTMLAreaElement>() ||
+        subject.is::<HTMLAnchorElement>() ||
+        subject.is::<HTMLFormElement>())
+    {
+        return None;
+    }
+    if subject.has_attribute(&local_name!("target")) {
+        return Some(
+            subject
+                .get_attribute(&ns!(), &local_name!("target"))
+                .unwrap()
+                .Value(),
+        );
+    }
+    if (document_from_node(subject).base_element())
+        .unwrap()
```

Comment:
We don't want to panic if there is no `<base>`, so let's do this:
```rust
document_from_node(subject)
    .base_element()
    .map(|base| base.upcast::<Element>())
    .and_then(|element| {
        if element.has_attribute(&local_name!("target")) {
            Some(element.get_string_attribute(&local_name!("target")))
        } else {
            None
        }
    })
```

Reply:
I get an error code E0515 about not being able to return a value referencing function parameter 'base'

Reply:
@jdm Would using a match statement instead of unwrapping the base_element() value eliminate the chance for a panic?

Reply:
It would. It may also be possible to do `.base_element().as_ref()`, or remove the `map` and do `let element = element.upcast::<Element>();` inside of the and_then.